database/multistmt/parse.go

```diff
@@ -98,6 +137,9 @@ func Parse(reader io.Reader, delimiter []byte, maxMigrationSize int, replacement
 							s1 := strings.ReplaceAll(string(c1), "<SCHEMA_NAME>", replacementStatement)
 							c1 = []byte(s1)
 						}
+						// in the future this could be the place to run statements
+						//instead of keeping them as an array(
+						//the array subverts the streaming intention of this reader)
```

Comment:
Does this mean we don't actually stream the statements? Do they all end up stored in memory?

Reply:
Yes, that's why I left that comment, we stream the bytes, then accumulate them in an array and then run them. I think it's a trivial change to remove the array and execute the statements during streaming.